google/validation.go

```diff
@@ -0,0 +1,24 @@
+package google
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	"regexp"
+)
+
+func validateGCPName(v interface{}, k string) (ws []string, errors []error) {
```

Comment:
`validateGCPName` is implicitly a `schema.SchemaValidationFunc` because of its signature; we want `validateRegexp` to validate a against a variable regex and not a fixed one like `validateGCPName`, so we pass that regex into a function that makes a `schema.SchemaValidationFunc` for us.

There are similar functions to `validateRegexp` living in the core Terraform repo's [helper/validation package](https://github.com/hashicorp/terraform/blob/master/helper/validation/validation.go).